File: tests/machine/x/java/min_max_builtin.java
import java.util.*;
public class MinMaxBuiltin {
	static List<Integer> nums = new ArrayList<>(Arrays.asList(3, 1, 4));
	public static void main(String[] args) {
	System.out.println(nums.stream().mapToInt(n -> ((Number)n).intValue()).min().orElse(Integer.MAX_VALUE));
	System.out.println(nums.stream().mapToInt(n -> ((Number)n).intValue()).max().orElse(Integer.MIN_VALUE));
	}
}

File: archived/tests/transpiler/x/pas/min_max_builtin.pas
{$mode objfpc}
program Main;
function min(xs: array of integer): integer;
var i, m: integer;
begin
  if Length(xs) = 0 then begin min := 0; exit; end;
  m := xs[0];
  for i := 1 to High(xs) do if xs[i] < m then m := xs[i];
  min := m;
end;
function max(xs: array of integer): integer;
var i, m: integer;
begin
  if Length(xs) = 0 then begin max := 0; exit; end;
  m := xs[0];
  for i := 1 to High(xs) do if xs[i] > m then m := xs[i];
  max := m;
end;
var
  nums: array of integer;
begin
  nums := [3, 1, 4];
  writeln(min(nums));
  writeln(max(nums));
end.
